requests-for-comments/2024-01-23-project-environments.md

```diff
@@ -0,0 +1,189 @@
+
+	They run dev and production separately. The user suggested *sharing* dashboards between projects, rather than copying.
+
+- [Insight & dashboard copying/templates + feature flag copying + sharing event definitions - D ZEN-3433](https://posthoghelp.zendesk.com/agent/tickets/3433)
+
+	A lot of useful feedback, but insights & dashboards take the cake. They want to keep development, staging and production separate, with an easy way to "promote" dashboards/insights to the higher level once ready.
+
+	They also highlighted a few other things as annoying to replicate between projects manually:
+
+	- feature flags
+	- actions, specifically webhook message formats
+	- some project settings, specifically person display name, webhook integration, group types
+
+	They also noted it's annoying how switching projects from a dashboard takes you back to the homepage. Sadly, this is currently the only thing we can do, because a dashboard is strictly accessible in only one project.
+
+- [Sharing event definitions - P ZEN-5939](https://posthoghelp.zendesk.com/agent/tickets/5939) and [ZEN-5937](https://posthoghelp.zendesk.com/agent/tickets/5937) (two from one org)
+
+	They don't want event and property definitions to get out of sync between staging and production projects, since they intend the shape of data to be the same for all environments. 
+
+- [Sharing event definitions - I ZEN-4637](https://posthoghelp.zendesk.com/agent/tickets/4637)
+
+	Same as above, with a focus on event descriptions.
+
+- [Experiment copying - S ZEN-8439](https://posthoghelp.zendesk.com/agent/tickets/8439)
+
+	They would like to copy experiments between staging and production. This is framed as "would be nice" though.
+
+- [Feature flag copying - O ZEN-10377](https://posthoghelp.zendesk.com/agent/tickets/10377)
+
+	Wanted to copy flags from dev to production.
+
+- [Feature flag copying - K ZEN-10652](https://posthoghelp.zendesk.com/agent/tickets/10652)
+
+	Same as above.
+
+#### Issues
+
+- [Using projects as environments: syncing actions/insights/dashboards/playlists etc. #13418](https://github.com/PostHog/posthog/issues/13418)
+
+	5 users have chimed in to express their need for a better UX around environments, and 11 more have upvoted a comment suggesting handling of "environments". All commenters mention insights and dashboards as a problem. Feature flags and event/property definitions are also mentioned.
+
+- [Copy / Move dashboards across projects #19476](https://github.com/PostHog/posthog/issues/19476)
+
+	3 users commented to express their need, one specifically mentioned dashboards, insights, and actions as currently problematic.
+
+#### User interviews
+
+- [Feature flag & experiment & dashboard copying - R, by Annika](https://posthog.slack.com/archives/C011L071P8U/p1671640760520999)
+
+	They run development and staging, same as customers above they find manual replication tedious and error-prone.
+
+### Summarizing needs
```

Comment:
Could we write out the requirements (must have vs nice to have) more explicitly here, before jumping into discussing solutions.

Furthermore, could we separate the requirements we have for the backend side vs the UI.

Here's a starting point for the requirements:
1. events, people, groups, recordings, surveys are separate and don't need sharing across projects
3. different features/products can have different sharing configurations (e.g. one might want that all dashboards are shared across all projects, but feature flags are not shared across dev and prod projects).
4. we don't duplicate data in the backend nor run syncs across projects
5. UI doesn't require use to individually set sync for each new item they create (note that in the backend we might set something always based on a higher level configuration)
...

Reply:
Fair, some requirements I kind of only "revealed" n later sections (reflecting what the process of writing the RFC looked like 😄). Consolidated in this section. Let me know if it looks like I've missed anything